cases/services/tag_enricher.py

```
@@ -0,0 +1,1349 @@
+    try:
+        for url in doc_urls:
+            tmp_path = None
+            try:
+                suffix = os.path.splitext(urlparse(url).path.lower())[1] or ".tmp"
+                tmp = tempfile.NamedTemporaryFile(suffix=suffix, delete=False)
+                tmp_path = tmp.name
+                tmp.close()
+
+                req = urllib.request.Request(
+                    url, headers={"User-Agent": "JawafdehiAPI/1.0"}
+                )
```

Ruff (0.15.12)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)